pandas/core/indexes/datetimes.py

```diff
@@ -771,7 +771,7 @@ def _maybe_cast_slice_bound(self, label, side: str, kind):
 
     def _get_string_slice(self, key: str):
         freq = getattr(self, "freqstr", getattr(self, "inferred_freq", None))
-        parsed, reso_str = parsing.parse_time_string(key, freq)
+        parsed, reso_str = parsing.parse_time_string(str(key), freq)
```

Comment:
can you try doing this higher in the stack as this *should* already be a string here

Reply:
sure, I moved it one level higher in the stack